lib/ansible/modules/system/listen_ports_facts.py

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/python
+# encoding: utf-8
+
+# (c) 2016, Nathan Davison <ndavison85@gmail.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: listen_ports_facts
+author:
+    - '"Nathan Davison (@ndavison)" <ndavison85@gmail.com>'
+version_added: "2.3"
+description:
+    - |
+      Gather facts on processes listening on TCP and UDP ports. Optionally provide a whitelist of TCP and/or UDP ports to gather facts on processes
+       that violate the whitelists.
+short_description: Gather facts on processes listening on TCP and UDP ports.
+options:
+  whitelist_tcp:
+    description:
+      - A list of TCP port numbers that are expected to have processes listening.
+    required: false
+    type: "list"
+  whitelist_udp:
+    description:
+      - A list of UDP port numbers that are expected to have processes listening.
+    required: false
+    type: "list"
+'''
+
+EXAMPLES = '''
+- name: gather facts on listening ports, and populate the tcp_listen_violations fact with processes listening on TCP ports that are not 22, 80, or 443.
+  listen_ports_facts:
+    whitelist_tcp:
+      - 22
+      - 80
+      - 443
+
+- name: TCP whitelist violation
+  debug:
+    msg: "TCP port {{item.port}} by pid {{item.pid}} violates the whitelist"
```

Comment:
It might be nicer if the documentation contains the description about return values.

Add `RETURN` docstring after `EXAMPLES`.

Here are the some examples of the modules with `RETURN` docstring.
- https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/ec2_lc_facts.py#L82
- https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/source_control/git.py#L232
- https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/files/acl.py#L139